web_src/js/features/dropzone.js

```diff
@@ -1,80 +1,131 @@
-import {POST} from '../modules/fetch.js';
+import {GET, POST} from '../modules/fetch.js';
 import {showErrorToast} from '../modules/toast.js';
+import {createElementFromHTML, createElementFromObject} from '../utils/dom.js';
 
 const {csrfToken, i18n} = window.config;
 
-export async function createDropzone(el, opts) {
+async function createDropzone(el, opts) {
   const [{Dropzone}] = await Promise.all([
     import(/* webpackChunkName: "dropzone" */'dropzone'),
     import(/* webpackChunkName: "dropzone" */'dropzone/dist/dropzone.css'),
   ]);
   return new Dropzone(el, opts);
 }
 
-export function initGlobalDropzone() {
-  for (const el of document.querySelectorAll('.dropzone')) {
-    initDropzone(el);
-  }
+function addCopyLink(file) {
+  // Create a "Copy Link" element, to conveniently copy the image or file link as Markdown to the clipboard
+  // The "<a>" element has a hardcoded cursor: pointer because the default is overridden by .dropzone
+  const copyLinkEl = createElementFromHTML(`
+<div class="tw-text-center">
+  <a href="#" class="tw-cursor-pointer">${svg('octicon-copy', 14)} Copy link</a>
+</div>`);
+  copyLinkEl.addEventListener('click', async (e) => {
+    e.preventDefault();
+    let fileMarkdown = `[${file.name}](/attachments/${file.uuid})`;
+    if (file.type?.startsWith('image/')) {
+      fileMarkdown = `!${fileMarkdown}`;
+    } else if (file.type?.startsWith('video/')) {
+      fileMarkdown = `<video src="/attachments/${htmlEscape(file.uuid)}" title="${htmlEscape(file.name)}" controls></video>`;
+    }
+    const success = await clippie(fileMarkdown);
+    showTemporaryTooltip(e.target, success ? i18n.copy_success : i18n.copy_error);
+  });
+  file.previewTemplate.append(copyLinkEl);
 }
 
-export function initDropzone(el) {
-  const $dropzone = $(el);
-  const _promise = createDropzone(el, {
-    url: $dropzone.data('upload-url'),
+/**
+ * @param {HTMLElement} dropzoneEl
+ */
+export async function initDropzone(dropzoneEl) {
+  if (!dropzoneEl) return null;
+
+  const listAttachmentsUrl = dropzoneEl.closest('[data-attachment-url]')?.getAttribute('data-attachment-url');
+  const removeAttachmentUrl = dropzoneEl.getAttribute('data-remove-url');
+  const attachmentBaseLinkUrl = dropzoneEl.getAttribute('data-link-url');
+
+  let disableRemovedfileEvent = false; // when resetting the dropzone (removeAllFiles), disable the "removedfile" event
+  let fileUuidDict = {}; // to record: if a comment has been saved, then the uploaded files won't be deleted from server when clicking the Remove in the dropzone
+  const dzInst = await createDropzone(dropzoneEl, {
+    url: dropzoneEl.getAttribute('data-upload-url'),
     headers: {'X-Csrf-Token': csrfToken},
-    maxFiles: $dropzone.data('max-file'),
-    maxFilesize: $dropzone.data('max-size'),
-    acceptedFiles: (['*/*', ''].includes($dropzone.data('accepts'))) ? null : $dropzone.data('accepts'),
+    maxFiles: dropzoneEl.getAttribute('data-max-file'),
```

Comment:
```suggestion
    maxFiles: Number(dropzoneEl.getAttribute('data-max-file')) || null,
```

`null` means unrestricted as per dropzone docs and `0` makes no sense.

Reply:
That's the old code. I believe Dropzone works that way

Reply:
From : Remove jQuery `.attr` from the code comments (#30112)

![image](https://github.com/go-gitea/gitea/assets/2114189/c2fd9f62-d50a-4c41-82e7-fee51a36361f)

Reply:
To make code stricter and 100% follow the dropzone's document:  d957c1c8a94356028e5005a67ee1c4f5e6080f76

Reply:
Thanks. Not sure the helper function was really needed because my examples should already work in all cases (null, empty string, NaN), but it's good to pass numbers where numbers are expected, as previous code was likely a unsupported usage that may have only worked by coincidence.

Reply:
Your example doesn't work for non-existing `data-` attributes. It needs "null" but your code gives "0"

Reply:
It would work: `Number(null)` is `0` so `||` activates the fallback.

Reply:
fyi:

```
> Number(null)
0
> Number("")
0
> Number("foo")
NaN

Reply:
It is a unclear solution. 0 is 0, null is null.

And Infinity can't be right either.

I do not think introducing undefined/unclear behaviors is right.

Reply:
NaN also can't be right. That's my my function throws an errors.

Fail fast to catch bugs.

Reply:
Reverted all 15ce3f20e1628f8554c0cb253f062911020d9400